aspects/aspects.go

```diff
@@ -441,6 +465,33 @@ func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
 				value:       val,
 			})
 		}
+
+		// prune this match's path from the value so that we can check whether
+		// anything is left at the end
+		copyValue, err = prunePathInValue(match.suffixParts, copyValue)
```

Comment:
can't the same suffixParts be used many times?  should we remember them and prune them at the end? before checking?

Reply:
It's possible although unlikely because it implies the requests are the same (which we only allow if they're all write-only). I've now changed it to accumulate all the paths first as you mentioned. The commit also refactors it into its own function for added clarity